build-logic/src/main/kotlin/config-shade.gradle.kts

```diff
@@ -0,0 +1,41 @@
+import com.github.jengelman.gradle.plugins.shadow.tasks.ShadowJar
+import gradle.kotlin.dsl.accessors._d73e0aad27c892ff134862f94ec3182c.build
+import gradle.kotlin.dsl.accessors._d73e0aad27c892ff134862f94ec3182c.implementation
+import gradle.kotlin.dsl.accessors._d73e0aad27c892ff134862f94ec3182c.jar
+import org.gradle.kotlin.dsl.creating
+
+plugins {
+    id("com.github.johnrengelman.shadow")
+}
+
+val shade: Configuration by configurations.creating
+configurations.implementation { extendsFrom(shade) }
+
+tasks {
+    val shadowJar by getting(ShadowJar::class) {
+        archiveClassifier.set("all")
+
+        configurations = listOf(shade)
+        exclude("META-INF/*.SF", "META-INF/*.DSA", "META-INF/*.RSA", "OSGI-INF/**", "*.profile", "module-info.class", "ant_tasks/**")
+
+        arrayOf(
+            "com.google",
+            "org.objectweb.asm",
+            "org.spongepowered",
+            "com.grappenmaker.mappings",
+        ).forEach { pkg ->
+            val lastPackageIndex = pkg.lastIndexOf('.')
+            val lastPackage = if (lastPackageIndex != -1) pkg.substring(lastPackageIndex + 1) else pkg
+            val relocated = "net.weavemc.loader.shaded.${lastPackage}"
```

Comment:
> but I see no reason to just keep using the old package name

You mean you see no reason not to? I mainly change this to prevent pollution in the `net.weavemc` package-space but can be reverted